src/backend/distributed/commands/multi_copy.c

```diff
@@ -2251,8 +2253,33 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 static bool
 CitusCopyDestReceiverReceive(TupleTableSlot *slot, DestReceiver *dest)
 {
+	bool result = false;
 	CitusCopyDestReceiver *copyDest = (CitusCopyDestReceiver *) dest;
 
+	PG_TRY();
+	{
+		result = CitusSendTupleToPlacements(slot, copyDest);
```

Comment:
Moved all of ex-`CitusCopyDestReceiverReceive()` to `CitusSendTupleToPlacements()`, so it is easier to surround it with `PG_TRY()`.